pkgs/applications/radio/gnuradio/pkgs/srcs.nix

```diff
@@ -0,0 +1,112 @@
+{ lib
+, ver
+, pkgs
+}:
+
+let
+  knownSrcs = {
```

Comment:
Using the same code for the derivation but exchanging only the sources between 3.7 and 3.8 relies on the assumption that the newer version does not require any modifications to derivation. How likely is it that this works?

Reply:
Good question. If you'll look at the modified expressions of the packages, you'll see there's isn't much there at all (besides osmosdr - the most complicated one). The reason they got so simplified is because they are built with `gnuradio.callPackage`'s `mkDerivation` which wraps `stdenv.mkDerivation` with `nativeBuildInputs` and `buildInputs` that are used in the unwrapped gnuradio's derivation - and that usually sums up all that is needed for these packages.

If modifications are needed for a gnuradio package based on the version of gnuradio itself, it's possible to import `gnuradio` in the expression of the package and add conditionals based on `gnuradio.versionAttrs`.

I checked (not with `nixpkgs-review`, but manually) that all attributes that hydra will build, build. For instance the attribute `gnuradioMinimal.pkgs.rds` doesn't build because rds 3.8 needs it's gnuradio (3.8) to be built with swig and python support which `gnuradioMinimal` is built without.

Reply:
OK, let's give it try.  The osmosdr derivation does not look that bad. 
The Python ecosystem has the same problems to handle. There one can disable a library for a specific version: `disabled = !isPy3k;` Can we do something similar?

Reply:
Our situation is a bit different. We want to have gnuradio plugins compiled for different gnuradio versions, but all of the upstream authors of these plugins had to release new versions that should work with gr-3.8. They didn't release these versions in a manner compatible with both 3.8 3.7 at the same time. Look at the end of `./pkgs/srcs.nix` to see how we "disable" certain plugins for certain gnuradio packages.

Reply:
OK, I see.